components/patient-data-sharing/push-api/src/test/java/org/phenotips/data/push/internal/DefaultPushServerConfigurationResponseTest.java

```diff
@@ -0,0 +1,138 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.data.push.internal;
+
+import net.sf.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.xwiki.component.manager.ComponentLookupException;
+
+import java.util.Set;
+import java.util.TreeSet;
+
+public class DefaultPushServerConfigurationResponseTest {
+
+    private DefaultPushServerConfigurationResponse defaultPushServerConfigurationResponse;
+    private JSONObject serverResponse;
+
+    @Before
+    public void setUp()
+    {
+        serverResponse = new JSONObject();
+        defaultPushServerConfigurationResponse = new
+                DefaultPushServerConfigurationResponse(serverResponse);
+    }
+
+    @Test
+    public void getRemoteUserGroupsTest() throws ComponentLookupException
+    {
+        // populate a JSONObject
+        for(int i = 0; i < 10; i++) {
+            serverResponse.accumulate("user_groups", i);
+        }
+        Set<String> expected = new TreeSet<String>();
+        for(int i = 0; i < 10; i++) {
+            expected.add(String.valueOf(i));
+        }
+        Assert.assertEquals(expected, defaultPushServerConfigurationResponse.getRemoteUserGroups());
+    }
+
+    @Test
+    public void getRemoteUserGroupsNullTest()
+    {
+        Assert.assertNull(defaultPushServerConfigurationResponse.getRemoteUserGroups());
+    }
+
+    @Test
+    public void getRemoteAcceptedPatientFieldsTest()
+    {
+        // populate a JSONObject
+        for(int i = 0; i < 10; i++) {
+            serverResponse.accumulate("accepted_fields", i);
+        }
+        Set<String> expected = new TreeSet<String>();
+        for(int i = 0; i < 10; i++) {
+            expected.add(String.valueOf(i));
+        }
+        Assert.assertEquals(expected, defaultPushServerConfigurationResponse.getRemoteAcceptedPatientFields());
+        Assert.assertEquals(expected, defaultPushServerConfigurationResponse.getRemoteAcceptedPatientFields("string"));
+    }
+
+    @Test
+    public void getRemoteAcceptedPatientFieldsNullTest()
+    {
+        Assert.assertNull(defaultPushServerConfigurationResponse.getRemoteAcceptedPatientFields());
+        Assert.assertNull(defaultPushServerConfigurationResponse.getRemoteAcceptedPatientFields("string"));
+    }
+
+    @Test
+    public void getPushableFieldsTest()
+    {
+        // populate a JSONObject
+        for(int i = 0; i < 10; i++) {
+            serverResponse.accumulate("accepted_fields", i);
+        }
+        Set<String> expected = new TreeSet<String>();
+        for(int i = 0; i < 10; i++) {
+            expected.add(String.valueOf(i));
+        }
+        Assert.assertEquals(expected, defaultPushServerConfigurationResponse.getRemoteAcceptedPatientFields());
```

Comment:
this test is an exact copy of `getRemoteAcceptedPatientFieldsTest`, I guess the intention was to implement alike